core/src/main/java/io/questdb/griffin/RecordToRowCopierUtils.java

```diff
@@ -0,0 +1,744 @@
+                            asm.invokeInterface(wPutInt, 2);
+                            break;
+                        case ColumnType.LONG:
+                            asm.i2l();
+                            asm.invokeInterface(wPutLong, 3);
+                            break;
+                        case ColumnType.DATE:
+                            asm.i2l();
+                            asm.invokeInterface(wPutDate, 3);
+                            break;
+                        case ColumnType.TIMESTAMP:
+                            asm.i2l();
+                            asm.invokeInterface(wPutTimestamp, 3);
+                            break;
+                        case ColumnType.FLOAT:
+                            asm.i2f();
+                            asm.invokeInterface(wPutFloat, 2);
+                            break;
+                        case ColumnType.DOUBLE:
+                            asm.i2d();
+                            asm.invokeInterface(wPutDouble, 3);
+                            break;
+                        default:
+                            asm.invokeInterface(wPutByte, 2);
+                            break;
+                    }
+                    break;
+                case ColumnType.SHORT:
+                    asm.invokeInterface(rGetShort);
+                    switch (toColumnTypeTag) {
+                        case ColumnType.BYTE:
+                            asm.invokeStatic(implicitCastShortAsByte);
+                            asm.invokeInterface(wPutByte, 2);
+                            break;
+                        case ColumnType.INT:
+                            asm.invokeInterface(wPutInt, 2);
+                            break;
+                        case ColumnType.LONG:
+                            asm.i2l();
+                            asm.invokeInterface(wPutLong, 3);
+                            break;
+                        case ColumnType.DATE:
+                            asm.i2l();
+                            asm.invokeInterface(wPutDate, 3);
+                            break;
+                        case ColumnType.TIMESTAMP:
+                            asm.i2l();
+                            asm.invokeInterface(wPutTimestamp, 3);
+                            break;
+                        case ColumnType.FLOAT:
+                            asm.i2f();
+                            asm.invokeInterface(wPutFloat, 2);
+                            break;
+                        case ColumnType.DOUBLE:
+                            asm.i2d();
+                            asm.invokeInterface(wPutDouble, 3);
+                            break;
+                        default:
+                            asm.invokeInterface(wPutShort, 2);
+                            break;
+                    }
+                    break;
+                case ColumnType.BOOLEAN:
+                    asm.invokeInterface(rGetBool);
```

Comment:
Consider guarding the `toType` to be boolean and throw an error if it's not. that would make it consistent with for example the timestamp case: https://github.com/questdb/questdb/blob/3076151b42ae55b0817942b58ddda6727e980d34/core/src/main/java/io/questdb/griffin/RecordToRowCopierUtils.java#L313

Reply:
I am going to implement this in  the followup PR, which I already started 👍